feature_engine/selection/probe_feature_selector.py

```diff
@@ -0,0 +1,233 @@
+            )
+
+        if not (0 <= cut_rate <= 1):
+            raise ValueError(
+                "cut_rate must range from 0 to 1, inclusively. "
+                f"Got {cut_rate} instead."
+            )
+
+        if not isinstance(cv, int) or not (cv > 0):
+            raise ValueError(
+                f"cv must be a positive integer. Got {cv} instead."
+            )
+
+        if not isinstance(n_iter, int) or not (n_iter > 0):
+            raise ValueError(
+                f"n_iter must be a positive integer. Got {n_iter} instead."
+            )
+        super().__init__(confirm_variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.distribution = distribution
+        self.cut_rate = cut_rate
+        self.cv = cv
+        self.n_iter = n_iter
+        self.random_state = random_state
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Create three random feature. Find initial model performance.
+        Sort features by importance.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
+
+        # find numerical variables
+        self.variables = _find_or_check_numerical_variables(X, None)
+
+        # if required exclude variables that are not in the input dataframe
+        # instantiates the 'variables_' attribute
+        self._confirm_variables(X)
+
+        # check that there is more than 1 variable
+        self._check_variable_number()
+
+        # save input features
+        self._get_feature_names_in(X)
+
+        # create probe feature distribution
+        self.probe_feature_data_ = self._generate_probe_feature(X.shape[0])
+
+        # get probe feature name
+        self.probe_feature_ = self.probe_feature_data_.columns
+
+        # merge X and probe feature
+        X_new = pd.concat([X, self.probe_feature_data_], axis=1)
+
+        # train model with all variables including the probe feature
+        model = cross_validate(
+            self.estimator,
+            X_new,
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+            return_estimator=True,
+        )
+
+        # Initialize a dataframe that will contain the list of the feature/coeff
+        # importance for each cross-validation fold
+        feature_importances_cv = pd.DataFrame()
+
+        # Populate the feature_importances_cv dataframe with columns containing
+        # the feature importance values for each model returned by the cross
+        # validation.
+        # There are as many columns as folds.
+        for i in range(len(model["estimator"])):
+            m = model["estimator"][i]
+            feature_importances_cv[i] = get_feature_importances(m)
```

Comment:
we are missing the step where we take the average of the importance across the different cv splits:

https://github.com/feature-engine/feature_engine/blob/a0106c968b14922d5123aaf51625fa66123dfad8/feature_engine/selection/base_recursive_selector.py#L182-L190

With this average, we are ready to select features. See for example here:
https://github.com/feature-engine/feature_engine/blob/a0106c968b14922d5123aaf51625fa66123dfad8/feature_engine/selection/base_recursive_selector.py#L182-L190

Reply:
I didn't average the feature importances because I thought the class is intended to compare the probe feature's importance to the other variables importances for multiple models.

This is the purpose of the `threshold` param. If we average the feature importances, there is only one feature importance comparison.

Reply:
I don't understand.

This is the logic:

dataset has 3 variables, A, B and C.

train say random forest, derive 3 feature importance values (the mean of the cv), so that each one would have 1 value, ie, A: 0,5, B:0.6, C:0.7

But, we added probe features, probA and probB, which were also used to train the random forest and have importance of 0.5 each. Hence, we select features B and C, because importance is higher than 0.5

If probA and probB have 0.4 and 0.5, then the threshold needs to be above 0.45. Which now that I am writing it, I realize that we do not really need the parameter threshold. If importance is greater than probe, then select, otherwise remove.

Reply:
I understand. My interpretation of @glevv's suggestion is different though. I understood that @glevv recommended testing the feature importance during every iteration, counting the results, then creating a proportion that is compared to the `threshold`.

I'll implement your approach unless advised otherwise. I may not get to it until later this week.

Reply:
I'd start simple. We just compare averages. If we want to include the uncertainty, later we can use the standard deviation. But let's get a minimum viable version first. And then we see if and when it is worth adding layers of complexity.